app/uk/gov/nationalarchives/omega/editorial/services/EditSetService.scala

```diff
@@ -21,20 +21,22 @@
 
 package uk.gov.nationalarchives.omega.editorial.services
 
+import cats.effect.unsafe.implicits.global
 import play.api.Logger
-import uk.gov.nationalarchives.omega.editorial.editSets
+import uk.gov.nationalarchives.omega.editorial.connectors.ApiConnector
 import uk.gov.nationalarchives.omega.editorial.models.EditSet
 
-import javax.inject.Singleton
+import javax.inject.{ Inject, Singleton }
+import scala.concurrent.Future
 
 @Singleton
-class EditSetService {
+class EditSetService @Inject() (apiConnector: ApiConnector) {
 
   private val logger: Logger = Logger(this.getClass)
 
-  def getEditSet(id: String): EditSet = {
+  def get(id: String): Future[EditSet] = {
     logger.info(s"The edit set id is $id ")
-    editSets.getEditSet()
+    apiConnector.getEditSet(id).unsafeToFuture()
```

Comment:
Use of `Future` should be restricted to the controller. This function should return `IO[EditSet]` and the `unsafeToFuture()` should be called at the end of controller method.

Reply:
That's moved now.